src/components/inputs/z-input/index.tsx

```diff
@@ -178,8 +186,28 @@ export class ZInput {
     };
   }
 
+  getNumberAttributes() {
+    if (this.type != InputTypeEnum.number) return;
+    return {
+      min: this.min,
+      max: this.max,
+      step: this.step,
+    };
+  }
+
+  getTelAttributes() {
+    if (this.type != InputTypeEnum.tel) return;
+    return {
+      pattern: this.pattern,
+    };
+  }
+
   renderInputText(type: InputTypeBean = InputTypeEnum.text) {
-    const attr = this.getTextAttributes();
+    const attr = {
```

Comment:
Questo metodo riceve in input un `type`, quindi non credo sia corretto che i metodi `getNumberAttributes` e `getTelAttributes` controllino `this.type`, le due cose non sono coerenti. Secondo me i due metodi andrebbero richiamati solo se `type` corrisponde al tipo corretto.

Reply:
Questa è l'alternativa che abbiamo implementato, però non siamo molto convinti che sia effettivamente un passo in avanti rispetto a prima. Ci sembra forse più pulita la versione attuale.
Pensavi a qualcosa di diverso, rispetto a quello che abbiamo scritto qui di seguito?

```
renderInputText(type: InputTypeBean = InputTypeEnum.text) {
    const textAttr = this.getTextAttributes();

    const numberAttr =
      type == InputTypeEnum.number ? this.getNumberAttributes() : null;

    const hasPattern =
      type == InputTypeEnum.password ||
      type == InputTypeEnum.text ||
      type == InputTypeEnum.tel ||
      type == InputTypeEnum.search ||
      type == InputTypeEnum.url ||
      type == InputTypeEnum.email;
    const patternAttr = hasPattern ? this.getPatternAttribute() : null;

    const attr = { ...textAttr, ...numberAttr, ...patternAttr };
}
```

Reply:
O in questo modo, oppure passando il type ai metodi getNumberAttributes() e getTelAttributes() e gestendo il type dentro (com'è fatto adesso, però col valore in input), come preferite.

Reply:
ah ok, avevamo fatto nel secondo modo ma eravamo tornati indietro! :)